diff -u pari-2.3.4/src/language/default.c pari-2.3.4.new/src/language/default.c
--- pari-2.3.4/src/language/default.c	2008-01-16 09:46:36 -0600
+++ pari-2.3.4.new/src/language/default.c	2009-06-17 12:34:55 -0500
@@ -15,6 +15,7 @@
 #include "pari.h"
 #include "paripriv.h"
 #include "anal.h"
+#undef HAS_STRFTIME
 
 #ifdef HAS_STRFTIME
 #  include <time.h>
diff -u pari-2.3.4/src/language/es.c pari-2.3.4.new/src/language/es.c
--- pari-2.3.4/src/language/es.c	2008-01-16 09:46:36 -0600
+++ pari-2.3.4.new/src/language/es.c	2009-06-17 03:02:11 -0500
@@ -2573,7 +2573,9 @@
 #if defined(UNIX) || defined(__EMX__)
 #  include <fcntl.h>
 #  include <sys/stat.h>
+#  ifndef __MINGW32__
 #  include <pwd.h>
+#  endif
 #  ifdef __EMX__
 #    include <process.h>
 #  endif
@@ -2911,7 +2913,7 @@
 static char *
 _expand_tilde(const char *s)
 {
-#if !defined(UNIX) && !defined(__EMX__)
+#if (!defined(UNIX) && !defined(__EMX__)) || defined(__MINGW32__)
   return pari_strdup(s);
 #else
   struct passwd *p;
@@ -3547,7 +3549,11 @@
   return id < 0 || close(id);
 }
 static int
+#  ifdef __MINGW32__
+pari_dir_exists(const char *s) { return mkdir(s); }
+#  else
 pari_dir_exists(const char *s) { return mkdir(s, 0777); }
+#  endif
 #else
 static int
 pari_file_exists(const char *s) { return 0; }
@@ -3632,7 +3638,7 @@
   size_t lpre, lsuf;
 
   pre = pari_tmp_dir();
-#ifdef UNIX
+#if defined(UNIX) && !defined(__MINGW32__) 
   sprintf(suf,".%ld.%ld", (long)getuid(), (long)getpid());
 #else
   sprintf(suf,".gpa");
Only in pari-2.3.4.new/src/language: es.c~
diff -u pari-2.3.4/src/language/init.c pari-2.3.4.new/src/language/init.c
--- pari-2.3.4/src/language/init.c	2008-03-04 06:29:45 -0600
+++ pari-2.3.4.new/src/language/init.c	2009-06-17 03:10:56 -0500
@@ -18,6 +18,9 @@
 /*                INITIALIZING THE SYSTEM, ERRORS                  */
 /*                                                                 */
 /*******************************************************************/
+#define ftime _ftime32
+#define time _time32
+
 #include <string.h>
 #include "pari.h"
 #include "paripriv.h"
